spark/v3.3/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestRewriteDataFilesProcedure.java

```diff
@@ -69,7 +69,7 @@ public void testZOrderSortExpression() {
   public void testRewriteDataFilesInEmptyTable() {
     createTable();
     List<Object[]> output = sql("CALL %s.system.rewrite_data_files('%s')", catalogName, tableIdent);
-    assertEquals("Procedure output must match", ImmutableList.of(row(0, 0)), output);
+    assertEquals("Procedure output must match", ImmutableList.of(row(0, 0, 0L)), output);
```

Comment:
These changes are the only ones that i'm worried about. This will be *very* brittle. We cannot guarantee that other random changes won't change the file sizes, like every parquet version change, or compression library upgrade or who knows what. Let's make this either a soft check OR programmatically figure out the size.

So either
(FileSize is around X)
or

((SELECT sum(file_size_in_bytes) from table.files)) should be RewriteResult.sizeInBytes)

Reply:
I've updated checks to make sure it's >= 0

Reply:
I've updated checks to compare results with programmatically determined file sizes